### PR TITLE
[#48] Canvas Multiple Character Drawing

### PR DESCRIPTION
**Pull request by xeoh**
See [#48].

### CODE REVIEW

**Review by bimak-server — APPROVED** [review]
👍

File: app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java [hunk 1]
```diff
@@ -98,48 +106,106 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
...
+    regions.clear();
+
+    // ratio = (x2 - x1) / (y2 - y1)
+    double regionRatio = (Locator.ORIGIN_REGION.getMaxX() - Locator.ORIGIN_REGION.getMinX())
+        / (Locator.ORIGIN_REGION.getMaxY() - Locator.ORIGIN_REGION.getMinY());
+
+    // getting one region's size
+    double gap = FeatureController.getInstance().getGap();
+    int regionWidth = ptToCanvas(fontSize);
+    int gapWidthSize = (int)(regionWidth * gap);
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java]: `int gapWidthSize = (int) (regionWidth * gap);`

File: app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java [hunk 2]
```diff
@@ -98,48 +106,106 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
...
+    int regionHeight = (int)(regionWidth / regionRatio);
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java:168]: `int regionHeight = (int) (regionWidth / regionRatio);`

File: app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java [hunk 3]
```diff
@@ -98,48 +106,106 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
...
+    int gapHeightSize = (int)(regionHeight * lineMargin);
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java:169]: `int gapHeightSize = (int) (regionHeight * lineMargin)`

File: app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java [hunk 4]
```diff
@@ -98,48 +106,106 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
...
+
+    int maxCol = getWidth() / (regionWidth + gapWidthSize);
+
+    int locatorsCount = locators.size();
+    int rowCount = (int)(Math.ceil(locatorsCount / (double)maxCol));
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java:174]: int rowCount = (int) (Math.ceil(locatorsCount / (double) maxCol));

File: app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java [hunk 5]
```diff
@@ -98,48 +106,106 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
...
+            curHeight + regionHeight, curHeight);
+        regions.add(region);
+      }
+    }
+  }
+
+  private int ptToCanvas(double point) {
+    double inches = point / 72;
+    double xdpi = getResources().getDisplayMetrics().xdpi;
+    return (int)(inches * xdpi);
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/mylittlefont/FontCanvasView.java]: `return (int) (inches * xdpi);`

**Review by hangpark — CHANGES_REQUESTED** [review]
(Coding convention) Spacing policy between casting type and value should be consistent.

**Review by hangpark — APPROVED** [review]

**Pull request merged**